### PR TITLE
community[minor]: Add support for bedrock guardrails and trace

### PR DESCRIPTION
**Pull request by QuinnGT**
**Description:**

Added support for AWS Bedrock Guardrails and Trace

**Update**

- Revised bedrock chat and util to use the correct headers for guardrail and trace
- Updated bedrock test to test using guardrails
- Updated examples to show better available details in use of bedrock

Test results:
```
 PASS  src/chat_models/tests/chatbedrock.int.test.ts (22.203 s)
  ✓ Test Bedrock chat model Generating search queries: Command-r (916 ms)
  ✓ Test Bedrock chat model: Command-r (1189 ms)
  ✓ Test Bedrock chat model: Command-r (1875 ms)
  ✓ Test Bedrock chat model streaming: Command-r (2099 ms)
  ✓ Test Bedrock chat model streaming: Command-r (1701 ms)
  ✓ Test Bedrock chat model HandleLLMNewToken: Command-r (1077 ms)
  ✓ Test Bedrock chat model: Mistral-7b-instruct (681 ms)
  ✓ Test Bedrock chat model streaming: Mistral-7b-instruct (3563 ms)
  ✓ Test Bedrock chat model HandleLLMNewToken: Mistral-7b-instruct (1343 ms)
  ✓ Test Bedrock chat model: Claude-3 (2194 ms)
  ✓ Test Bedrock chat model streaming: Claude-3 (2673 ms)
  ✓ Test Bedrock chat model HandleLLMNewToken: Claude-3 (2682 ms)
  ○ skipped Test Bedrock base chat model: amazon.titan-text-express-v1
```

### CODE REVIEW

**Comment by QuinnGT** [comment]
I'm unable to get the test to pass. Not sure what I'm missing but I'll mark this as draft for now.

```
  ● Test Bedrock chat model HandleLLMNewToken: Claude-3

    Failed to access underlying url 'bedrock-runtime.us-west-2.amazonaws.com': got 400 Bad Request: {"message":"Malformed input request: #: subject must not be valid against schema {\"required\":[\"messages\"]}#: extraneous key [trace] is not permitted#: extraneous key [guardrailVersion] is not permitted#: extraneous key [guardrailIdentifier] is not permitted, please reformat your input and try again."}

      380 |
      381 |     if (response.status < 200 || response.status >= 300) {
    > 382 |       throw Error(
          |             ^
      383 |         `Failed to access underlying url '${endpointHost}': got ${
      384 |           response.status
      385 |         } ${response.statusText}: ${await response.text()}`

      at BedrockChat.Error [as _streamResponseChunks] (src/chat_models/bedrock/web.ts:382:13)
      at BedrockChat.chunk [as _generate] (src/chat_models/bedrock/web.ts:235:24)
          at async Promise.allSettled (index 0)
      at BedrockChat._generateUncached (../../langchain-core/dist/language_models/chat_models.js:169:29)
      at BedrockChat.invoke (../../langchain-core/dist/language_models/chat_models.js:53:24)
      at Object.<anonymous> (src/chat_models/tests/chatbedrock.int.test.ts:317:20)

Test Suites: 1 failed, 1 total
Tests:       6 failed, 2 skipped, 6 passed, 14 total
Snapshots:   0 total
Time:        10.718 s
Ran all test suites matching /chatbedrock.int.test.ts/i.
```

**Comment by QuinnGT** [comment]
All tests passed when guardrails are or are not in use. PR is ready for review.

File: libs/langchain-community/src/chat_models/bedrock/web.ts [hunk 1]
```diff
@@ -97,14 +101,92 @@ export function convertMessagesToPrompt(
  * Services (AWS). It uses AWS credentials for authentication and can be
```
[the remark this answers is not in the document]
Comment (QuinnGT) [reply libs/langchain-community/src/chat_models/bedrock/web.ts:101]: that is just example text, no function used here

File: libs/langchain-community/src/chat_models/tests/chatbedrock.int.test.ts [hunk 2]
```diff
@@ -156,28 +163,52 @@ async function testChatModel(
       credentials: {
```
[the remark this answers is not in the document]
Comment (QuinnGT) [reply libs/langchain-community/src/chat_models/tests/chatbedrock.int.test.ts:163]: Correct, this is a test function.

**Comment by jacoblee93** [comment]
Awesome, thank you! Looks reasonable, will pull and try it. May revert the int test region changes.

File: libs/langchain-community/src/chat_models/bedrock/web.ts [hunk 3]
```diff
@@ -89,6 +89,10 @@ export function convertMessagesToPrompt(
   throw new Error(`Provider ${provider} does not support chat.`);
 }
 
+interface BedrockCallOptions extends Partial<BaseChatModelParams> {
+  signal?: AbortSignal;
```
Comment (jacoblee93) [inline libs/langchain-community/src/chat_models/bedrock/web.ts:93]: This shouldn't be necessary actually

**Comment by jacoblee93** [comment]
Thank you!

**Pull request merged**